### DISCUSSION TITLE
Is it possible to trigger a workflow when someone adds / modifies / removes the label after pull request is created?

### DISCUSSION THREAD

**Discussion by meetatmoe** [woodpecker-ci/woodpecker#3882] [Request a Feature] [answered]
I know there is environment variable to pull the labels for current Pull request, there is no event for label change .

**Comment by anbraten** [comment]
There is the idea to add several events as mentioned here: https://github.com/woodpecker-ci/woodpecker/discussions/2174 

Atm a label update event is not implemented

**Reply by meetatmoe** [reply]
Any direction on where to implement in source code? I can try implementing it, but I'm new to Go. But worth a try.

**Reply by anbraten** [reply, marked as answer]
Sure you could give it a try. 

It's basically these steps:
- handle issue, pr webhooks in the forges
- add a new events to the code
  - would actually go for issue_created, issue_updated, issue_closed and pr_created, pr_updated, pr_closed (which already exists)
- the current labels of an issue / pr could be easily set as sth like CI_ISSUE_LABELS

To keep the changes easy and small for the beginning go with your actual use-case only. Other things can be added easily later on when we follow this plan.

**Reply by meetatmoe** [reply]
Cool, thanks!
